### PR TITLE
fix(clippy): Remove unnecessary clone/to_string calls

### PR DESCRIPTION
**Pull request by jalil-salame**
# Purpose

Implement clippy fixes that relate to allocations, by removing calls to `clone`, `to_string` and `format!` when unnecessary.

Sometimes `clone` is removed in favor of copy (`*my_var`), which does not reduce allocations (as clone would not have allocated anyways), but it makes it clear that no allocation is performed.

## Concerns

A lot of calls to `to_string` are passed directly to `format!`, ie. `format!("{}", e.to_string())`, is this intended?

## Tests

All tests pass locally (Linux).

### CODE REVIEW

File: lua-api-crates/json/src/lib.rs [hunk 1]
```diff
@@ -150,7 +150,7 @@ fn lua_value_to_json_value(value: LuaValue, visited: &mut HashSet<usize>) -> mlu
                         mlua::Error::FromLuaConversionError {
                             from: lua_type,
                             to: "value",
-                            message: Some(format!("while processing {key:?}: {}", e.to_string())),
+                            message: Some(format!("while processing {key:?}: {}", e)),
```
Comment (wez) [inline lua-api-crates/json/src/lib.rs:153]:
```suggestion
                            message: Some(format!("while processing {key:?}: {e}")),
```

File: lua-api-crates/share-data/src/lib.rs [hunk 2]
```diff
@@ -222,7 +222,7 @@ fn lua_value_to_gvalue_impl(value: LuaValue, visited: &mut HashSet<usize>) -> ml
                         mlua::Error::FromLuaConversionError {
                             from: lua_type,
                             to: "value",
-                            message: Some(format!("while processing {key:?}: {}", e.to_string())),
+                            message: Some(format!("while processing {key:?}: {}", e)),
```
Comment (wez) [inline lua-api-crates/share-data/src/lib.rs:225]:
```suggestion
                            message: Some(format!("while processing {key:?}: {e}")),
```

File: luahelper/src/lib.rs [hunk 3]
```diff
@@ -229,7 +229,7 @@ fn lua_value_to_dynamic_impl(
                         mlua::Error::FromLuaConversionError {
                             from: lua_type,
                             to: "value",
-                            message: Some(format!("while processing {key:?}: {}", e.to_string())),
+                            message: Some(format!("while processing {key:?}: {}", e)),
```
Comment (wez) [inline luahelper/src/lib.rs:232]:
```suggestion
                            message: Some(format!("while processing {key:?}: {e}")),
```

**Review by wez — COMMENTED** [review]
I think this looks good, thanks!
Couple of minor suggestions inline!

File: lua-api-crates/json/src/lib.rs (hunk 1 above)
Comment (jalil-salame) [reply lua-api-crates/json/src/lib.rs]: force-pushed fix

**Comment by jalil-salame** [comment]
Rebased on top of main and fixed comments, ready for review again.

**Comment by wez** [comment]
Thanks!

**Pull request merged**